### PR TITLE
Tools: Added git client hooks

### PR DESCRIPTION
**Pull request by gverni**
I've been playing with [git client hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) lately and I feel they are a pretty powerful to make sure some policies are reinforced before something is committed. The first thing I thought about is to run linter in the `pre-commit` phase. This hook will run the `lint` task before a commit is happening. If the linter exits with error then the commit is aborted. 

I know that we all run linter before we commit, and that `git commit --amend` is our best friend if we forget something 🤓. But on the other hand I feel this is another automation tool that can save some time for us and make sure that who contribute to this project is following proper guidelines. 

Also note that you can always bypass the check using `git commit --no-verify`. 

To test this feature: 
- Clone Sweetalert2
- Checkout this branch (`feat/git-client-hooks`)
- Run `npm` or `yarn` install. At the end of the install procedure you should see: 
```
> sweetalert2@8.8.3 postinstall 
> gulp install-git-client-hooks
```
- Change one of the file adding for example a `console.log()` statement
- Commit your changes. You should see
```
 git commit -am "Should be rejected"
(node:26515) ExperimentalWarning: The fs.promises API is experimental
[10:28:04] Using gulpfile ~/tmp/swal2/gulpfile.js
[10:28:04] Starting 'lint'...
...
  9:1  error  Unexpected console statement  no-console
```
And the commit is aborted (you can check the `git log` to make sure it's not there) 

This PR is not complete. TODO list: 

- [x] Tested on macOS with yarn and git version 2.6.4
- [x] Tested on macOS with npm and git version 2.6.4
- [x] Tested on win10 with npm and git version  2.18.0
- [x] Tested on win10 with yarn and git version  2.18.0
- [x] Tested on ubuntu 16.04 with npm and git version 2.7.4
- [ ] Tested on ubuntu 16.04 with yarn and git version 2.7.4
- [x] Remove the install hook task if executed in TravisCI and other CI / CD tools (I'm doing this PR to check what happens)
- [x] Add uninstall hooks task
- [x] Get the agreement from the maintainer and collaborators :smile:

Note: I decided to install the hook(s) in the `.git/hooks/` folder instead of using the `core.hooksPath` configuration. The reason is that this configuration is supported only by Git version >=2.9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,7 +105,8 @@
     "check:third-party": "npm run check:unpkg && npm run check:jsdelivr && npm run check:wappalyzer",
     "check:wappalyzer": "curl 'https://api.wappalyzer.com/lookup-basic/v1/?url=https%3A%2F%2Fsweetalert2.github.io' 2>&1 | grep --quiet 'SweetAlert2'",
     "check:unpkg": "curl --location 'https://unpkg.com/sweetalert2' 2>&1 | grep --quiet 'window.Swal'",
-    "check:jsdelivr": "curl --location 'https://cdn.jsdelivr.net/npm/sweetalert2' 2>&1 | grep --quiet 'window.Swal'"
+    "check:jsdelivr": "curl --location 'https://cdn.jsdelivr.net/npm/sweetalert2' 2>&1 | grep --quiet 'window.Swal'",
+    "postinstall": "./node_modules/.bin/gulp install-git-client-hooks"
```
Comment (limonte) [inline package.json]: `./node_modules/.bin/` can be omitted here, npm/yarn will look into `./node_modules/.bin/` by default.

**Comment by limonte** [comment]
I'm using the latest Git version (2.21.0) and can confirm that everything works nicely with Yarn and Ubuntu 18.04

**Comment by limonte** [comment]
I made this a PR to run Travis builds, I'll check everything on Windows machine later today.

**Comment by gverni** [comment]
Thank you so much @limonte. I'll check yarn on linux later as well (though I'm expecting it to work fine).

**Comment by gverni** [comment]
Tested on win10 (with both npm and yarn) 👍

**Pull request merged**